lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -361,8 +361,23 @@ static bool isStoreCopy(SILValue value) {
     SmallVector<SILValue, 4> roots;
     findGuaranteedReferenceRoots(source, /*lookThroughNestedBorrows=*/true,
                                  roots);
-    if (llvm::any_of(roots, [](SILValue root) {
-          return isa<BeginApplyInst>(root->getDefiningInstruction());
+    SSAPrunedLiveness liveness;
+    if (llvm::any_of(roots, [&liveness, storeInst](SILValue root) {
```

Comment:
fwiw, this is an example of where I find range-based for loops easier to read (rather than mentally factoring a closure with inverted an return value). Alternatively, you explicitly factor it into something like `isStoreCopyRootValid`

Reply:
ok.... you basically already did that in the following commit